toolchain/source/source_buffer.h

```diff
@@ -50,13 +55,24 @@ class SourceBuffer {
     return text_->getBuffer();
   }
 
+  [[nodiscard]] auto is_regular_file() const -> bool {
```

Comment:
I think we aren't typically doing nodiscard, we should really decide what to do about it.

```suggestion
  auto is_regular_file() const -> bool {
```

Reply:
This file uses `[[nodiscard]]` on all its other accessors, so I'd prefer to be locally consistent here.

Reply:
Okay, re-raising the question on #toolchain (I'd brought this issue up a couple weeks ago -- I think the main result here is going to be I remove it, and blame won't work as well)